torch/csrc/jit/tensorexpr/tensorexpr_init.cpp

```diff
@@ -508,77 +513,83 @@ void initTensorExprBindings(PyObject* module) {
           [](const LoopNest& self,
              ForPtr root,
              const std::vector<int>& indices) {
-            return self.getLoopAt(root, indices);
+            return self.getLoopAt(std::move(root), indices);
           },
           py::return_value_policy::reference)
       .def(
           "get_parent_loop",
-          [](const LoopNest& self, StmtPtr s) { return self.getParentLoop(s); },
+          [](const LoopNest& self, const StmtPtr& s) {
+            return self.getParentLoop(s);
+          },
           py::return_value_policy::reference)
       .def_static(
           "get_loop_stmts_in_loopnest",
-          [](ForPtr f, size_t num) {
+          [](const ForPtr& f, size_t num) {
             return LoopNest::getLoopStmtsInLoopNest(f, num);
           },
           py::return_value_policy::reference)
       .def(
           "split_with_tail",
-          [](ForPtr f, int factor) {
+          [](const ForPtr& f, int factor) {
             ForPtr inner = nullptr, tail = nullptr;
             LoopNest::splitWithTail(f, factor, &inner, &tail);
             return std::make_tuple(inner, tail);
           },
           py::return_value_policy::reference)
       .def(
           "split_with_mask",
-          [](ForPtr f, int factor) {
+          [](const ForPtr& f, int factor) {
             ForPtr inner = nullptr;
             LoopNest::splitWithMask(f, factor, &inner);
             return inner;
           },
           py::return_value_policy::reference)
       .def(
           "slice_head",
-          [](ForPtr f, int factor) {
+          [](const ForPtr& f, int factor) {
```

Comment:
Nit some of the makes_tuples should have their args moved into position for return.